dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -67,6 +83,52 @@ class TypedDataset[T](
     ): GroupedByManyOps[T, K, Out0] = new GroupedByManyOps[T, K, Out0](self, groupedBy)
   }
 
+  def join[A: TypedEncoder, B](
+    right: TypedDataset[A],
+    leftCol: TypedColumn[T, B],
```

Comment:
What's the advantage of doing this as opposed to defining a complete join expression, like

    a.join(b)(a.col(`aa) === b.col(`bb))

?